commit e95ef5d3f6bc60433883e1ef65dac747acd0bf1a
Author: Andre Detsch <adetsch@br.ibm.com>
Date:   Mon Apr 26 05:38:27 2010 +0000

    cxgb3: Wait longer for control packets on initialization
    
    In some Power7 platforms, when using VIOS (Virtual I/O Server), we
    need to wait longer for control packets to finish transfer during
    initialization.
    Without this change, initialization may fail prematurely.
    
    Signed-off-by: Wen Xiong <wenxiong@us.ibm.com>
    Signed-off-by: Andre Detsch <adetsch@br.ibm.com>
    Acked-by: Divy Le Ray <divy@chelsio.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/cxgb3/cxgb3_main.c b/drivers/net/cxgb3/cxgb3_main.c
index aced6c5..e3f1b85 100644
--- a/drivers/net/cxgb3/cxgb3_main.c
+++ b/drivers/net/cxgb3/cxgb3_main.c
@@ -439,7 +439,7 @@ static void free_irq_resources(struct adapter *adapter)
 static int await_mgmt_replies(struct adapter *adap, unsigned long init_cnt,
 			      unsigned long n)
 {
-	int attempts = 5;
+	int attempts = 10;
 
 	while (adap->sge.qs[0].rspq.offload_pkts < init_cnt + n) {
 		if (!--attempts)
